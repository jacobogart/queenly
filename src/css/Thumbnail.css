.ThumbnailImg {
  width: 100px;
  border-radius: 50%;
}