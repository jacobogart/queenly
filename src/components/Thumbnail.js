import React, { Component } from 'react';

// * CSS imports
import '../css/Thumbnail.css';

export default class Thumbnail extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  setResult = () => {
    this.props.selectResult(this.props.name);
  };

  render() {
    return (
      <figure 
        className="Thumbnail"
        onClick={this.setResult}
      >
        <img 
          className="ThumbnailImg"
          src={this.props.imgURL}/>
        <figcaption>{this.props.name}</figcaption> 
      </figure>
    )
  }
}

   