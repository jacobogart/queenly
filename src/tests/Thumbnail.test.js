import React from 'react';
import ReactDOM from 'react-dom';
import Thumbnail from '../components/Thumbnail.js';
import { shallow } from 'enzyme';

const mockResult = {
      "category": "drag show",
      "dayOfWeek": "Sudnay",
      "frequency": "weekly",
      "host": ["Chamilla Foxx"],
      "id": 2101,
      "imageURL": "https://pbs.twimg.com/media/D1Any4VWkAA86HL.jpg",
      "reoccuring": true,
      "startTime": [1400],
      "name": "Sunday Social"
    }

const mockSelectSearchResult = jest.fn();

describe('Thumbnail', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Thumbnail 
        imgURL={mockResult.imageURL}
        selectSearchResult={mockSelectSearchResult}
        name={mockResult.name}
        key={mockResult.id} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should have default state', () => {
    expect(wrapper.state()).toEqual({
      // Put state here
    });
  });
})