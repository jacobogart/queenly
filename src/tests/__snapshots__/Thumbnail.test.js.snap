// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should match snapshot 1`] = `
<figure
  className="Thumbnail"
  onClick={[Function]}
>
  <img
    className="ThumbnailImg"
    src="https://pbs.twimg.com/media/D1Any4VWkAA86HL.jpg"
  />
  <figcaption>
    Sunday Social
  </figcaption>
</figure>
`;
